### ISSUE TITLE
Electrum backend failure after running for a few days

### ISSUE THREAD

**Issue by erikarvstedt** [mempool/mempool#2246]
**Setup**: mempool with electrum backend (fulcrum), running [on NixOS](https://github.com/fort-nix/nixbitcoin.org/pull/44).
**Issue**: After the backend has run for a few days (mostly: 10 days+), electrum API queries fail with error `Connection to server lost, please retry`.
See it live:
https://mempool.nixbitcoin.org/address/1wiz18xYmhRX6xStj2b9t1rwWX4GKUgpv
```bash
curl -sSL https://mempool.nixbitcoin.org/api/address/1wiz18xYmhRX6xStj2b9t1rwWX4GKUgpv
# Connection to server lost, please retry
```

Checking with strace or tcpdump shows that there's zero TCP traffic between the mempool backend and fulcrum while the backend is processing the request.
The error disappears after restarting the backend. 
During the whole process, fulcrum runs fine and serves queries normally.

#### Versions

2.3.1
2.4.0-dev (`54931cb23ebfdf03219103801688932f7a5ecfd2`)

#### Discussion

See also: https://github.com/fort-nix/nix-bitcoin/pull/456#issuecomment-1173165473

If you're interested, we can give you access to the nodejs debug socket of our failed backend instance via SSH TCP tunneling.

**Comment by erikarvstedt** [comment]
The issue reappeared after running 35 days on rev `12c188266a67ebb7a358ed279605700cbbeb75cc`.
Has this been reported before by other users running the electrs backend?

**Comment by wiz** [comment]
@knorrium isn't there an issue with axios getting into a broken state?

**Comment by erikarvstedt** [comment]
@wiz, @knorrium, has this bug ever been reported before by other users running the electrum backend?
If no, I'll investigate on our Nix packaging side.

**Comment by wiz** [comment]
I thought we fixed it already, AFAIK you are the only person having the issue now

**Comment by wiz** [comment]
@knorrium I know you're still investigating this issue but I think this might be the same issue that was recently reported by @thebluematt where we aren't using axios everywhere and still using the NodeJS standard library HTTP client in some places which seems buggy or not working correctly - can you look into simply replacing the old HTTP client usage with axios and it might solve these issues?

**Comment by softsimon** [comment]
We've made 2 fixes in order to hopefully fix this in v2.5.

https://github.com/mempool/mempool/pull/3553
https://github.com/mempool/electrum-client/pull/4

**Comment by sampleblood** [comment]
I have encountered this bug too, after ten days or later，@mempool/electrum-client did not retry when failed, before failed, client retried when failed.

**Comment by erikarvstedt** [comment]
The issue seems fixed in v2.5.0. We're planning to add mempool to the main nix-bitcoin repo soon.

**Comment by erikarvstedt** [comment]
I can confirm that this bug is fixed.
Both [debug instances](https://github.com/fort-nix/nix-bitcoin/pull/505#issuecomment-1398733797) have now been running stably for 89 days.
